HST/get_program_info.py

```diff
@@ -0,0 +1,142 @@
+##########################################################################################
+# get_program_info/__init__.py
+#
+# get_program_info is the main function called in get_program_info pipeline task script.
+# It will do these actions:
+# - Download the proposal files via a web query.
+# - If these files are the same as the existing ones, return.
+# - Otherwise:
+#   - Rename each existing file by appending “-<ymdhms>” before its extension and moving
+#     it to a backups/ subdirectory.
+#   - Save the newly downloaded files.
+# - Generate the new program-info.txt with a possibly modified citation.
+##########################################################################################
+
+import fs.path
+import os
+import pdslogger
+import urllib.error
+import urllib.parse
+import urllib.request
+
+from citations import Citation_Information
+from hst_helper import (DOCUMENT_EXT,
+                        DOCUMENT_EXT_FOR_CITATION_INFO,
+                        PROGRAM_INFO_FILE)
+from hst_helper.fs_utils import (backup_file,
+                                 get_formatted_proposal_id,
+                                 get_program_dir_path)
+
+def get_program_info(proposal_id, download_dir=None, logger=None):
+    """Download proposal files and generate PROGRAM_INFO_FILE for the given proposal ID.
+
+    Input:
+        proposal_id:    a proposal id.
```

Comment:
Any function that returns a value should also have a "Returns:" section in the docstring.